msteams-platform/m365-apps/combine-office-add-in-and-teams-app.md

```diff
@@ -0,0 +1,618 @@
+---
+title: Add an Outlook Add-in to a Teams app
+description: Learn how to add an Outlook Add-in to a Teams app.
+ms.date: 01/11/2024
+ms.author: mosdevdocs
+author: rickki
+ms.topic: tutorial
+ms.localizationpriority: medium
+ms.subservice: m365apps
+---
+
+# Add an Outlook Add-in to a Teams app
+
+## Introduction
+
+Outlook Add-ins are web apps that extend the functionality of Outlook. Some of the things that can be done with an Outlook Add-in:
+
+- Read and write the content of email messages and meeting invitations, as well as responses, cancellations, and appointments.
+- Read properties of the user's mailbox.
+- Respond automatically to events, such as the sending of an email.
+- Integrate with external services including CRM and project management.
+- Add custom ribbon buttons or menu items to perform specific tasks.
+
+[Outlook Add-ins Overview](/office/dev/add-ins/outlook/outlook-add-ins-overview) which will give you information about what you can do with Outlook Add-ins, how they are structured, how they differ from older ways of extending Outlook, what platforms the add-in can run on (Windows, Mac, mobile, and the web), and how to get started creating one.
+
+Combining an Outlook Add-in with a Teams app enables scenarios that can't be satisfied with either of them separately. For example, consider an app that enables sales persons to insert a customized discount into an email for a customer, and then records the offer in a backend database. The app also enables the sales manager to view data about all the discounts that have been offered in a Teams tab. There is a sample of this app with animated screen shots at [Discount Offers](https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/tab-add-in-combined/nodejs).
+
+## Prerequisites
+
+To configure an Office Add-in as additional capability, you must meet the following conditions:
+
+- You have a Microsoft 365 account that includes Teams, or separate subscriptions to both Microsoft 365 and Teams, to test the application. For example, an *.onmicrosoft.com test account through the [Microsoft 365 Developer Program](/office/developer-program/microsoft-365-developer-program).
+- Your Microsoft 365 account has been added as an account in desktop Outlook. See [Add an email account to Outlook](https://support.microsoft.com/office/add-an-email-account-to-outlook-e9da47c4-9b89-4b49-b945-a204aeea6726)
+- To deploy the Teams app to Azure as described in the last section of this article, you need an Azure account with an active subscription. Create your free Azure account if you don't already have one by using the link [Free Azure account](https://azure.microsoft.com/free/).
+
```

Comment:
Should we add a prereq that you need an existing Teams app and Add-in that you want to package together? And then note that alternately, there's a completed sample available if you'd like to walk through the steps with that. Maybe even note that the following steps use the Contoso Add-in sample as an example.

Reply:
I'll add that you need an existing Teams app. But the tutorial has the reader create an Outlook add-in, so we can't really say they need to have one already. 

There is danger here. The process described in this tutorial has always been fragile. It has to assume that both the Teams app and the add-in have a certain folder structure and have certain configuration files with specific content. The Teams app must be one that was created with the latest version of Teams Toolkit. The add-in also must be created with the latest version of Teams Toolkit (or with a tool called Yo Office, provided that they make the "right" choices in that tool). 

In other words, this article doesn't give generic guidance for how to combine any old Teams app with any old Outlook add-in. We could have an article like that someday, but it would be exceedingly vague. There is an infinite number of ways that developers can structure the folders and source files for both a Teams app and an Office add-in. They don't all even use the same language. Some add-ins use TypeScript, others JavaScript, with concomitant changes in various config files. 

The hope is that developers will go through this tutorial and then be able to infer general principles about how to combine add-ins with Teams apps, and then apply those principles to their own Teams app and add-in. Preethika Kiruveedula was the DevX PM for Office at the time and can tell you more about the reasoning and history of this article.

Reply:
Gotcha, makes sense.